ui/app/appointments/controllers/manage/list/appointmentsListViewController.js

```diff
@@ -26,25 +27,39 @@ angular.module('bahmni.appointments')
-                {heading: 'APPOINTMENT_ADDITIONAL_INFO', sortInfo: 'additionalInfo', class: true, enable: true},
-                {heading: 'APPOINTMENT_CREATE_NOTES', sortInfo: 'comments', enable: true}];
             var init = function () {
                 $scope.searchedPatient = $stateParams.isSearchEnabled && $stateParams.patient;
                 $scope.startDate = $stateParams.viewDate || moment().startOf('day').toDate();
                 $scope.isFilterOpen = $stateParams.isFilterOpen;
+
             };
+            var buildTableHeader = function(){
+
+                $scope.tableInfo = [{heading: 'APPOINTMENT_PATIENT_ID', sortInfo: 'patient.identifier', enable: true},
+                                {heading: 'APPOINTMENT_PATIENT_NAME', sortInfo: 'patient.name', class: true, enable: true},
+                                {heading: 'APPOINTMENT_DATE', sortInfo: 'date', enable: true},
+                                {heading: 'APPOINTMENT_START_TIME_KEY', sortInfo: 'startDateTime', enable: true},
+                                {heading: 'APPOINTMENT_END_TIME_KEY', sortInfo: 'endDateTime', enable: true},
+                                {heading: 'APPOINTMENT_PROVIDER', sortInfo: 'provider.name', class: true, enable: true},
+                                {heading: 'APPOINTMENT_SERVICE_SPECIALITY_KEY', sortInfo: 'service.speciality.name', enable: $scope.enableSpecialities},
+                                {heading: 'APPOINTMENT_SERVICE', sortInfo: 'service.name', class: true, enable: true},
+                                {heading: 'APPOINTMENT_SERVICE_TYPE_FULL', sortInfo: 'serviceType.name', class: true, enable: $scope.enableServiceTypes},
+                                {heading: 'APPOINTMENT_STATUS', sortInfo: 'status', enable: true},
+                                {heading: 'APPOINTMENT_WALK_IN', sortInfo: 'appointmentKind', enable: true},
+                                {heading: 'APPOINTMENT_SERVICE_LOCATION_KEY', sortInfo: 'location.name', class: true, enable: true},
+                                {heading: 'APPOINTMENT_CREATE_NOTES', sortInfo: 'comments', enable: true}];
+                                 if($scope.additionalInfoColumns.length <= 0 ){
+                                    $scope.tableInfo.splice($scope.tableInfo.length - 1, 0,
+                                    {heading: 'APPOINTMENT_ADDITIONAL_INFO', sortInfo: 'additionalInfo', class: true, enable: true});
+                                }else{
+                                _.forEach($scope.additionalInfoColumns, function(value) {
+                                    var str = value.replace(/ +/g, "");
+                                    $scope.tableInfo.splice($scope.tableInfo.length - 1, 0,
+                                    {heading: 'APPOINTMENT_ADDITIONAL_INFO_COLUMN_'+str.toUpperCase(), sortInfo: value, class: true, enable: true});
+                                 });
+                                }
```

Comment:
Can you pull it into a method? Also, `$scope.additionalInfoColumns.length` can't be less than **0**

Reply:
Changed the check